apps/design/frontend/src/ballot_viewer.tsx

```diff
@@ -298,106 +54,142 @@ const Controls = styled.div`
+  background: ${({ theme }) => theme.colors.containerHigh};
+
+  /* Make sure overflow can scroll horizontally */
+  min-width: 0;
+`;
+
+const PdfContainer = styled.div`
   height: 100%;
-  background: ${({ theme }) => theme.colors.container};
+  display: flex;
+  flex-direction: column;
 `;
 
-const DebugPanelWrapper = styled.div`
-  position: absolute;
-  bottom: 0.5rem;
-  right: 0.75rem;
+const PdfDocumentScroller = styled.div`
+  overflow: auto;
+  flex: 1;
+  padding: 2rem;
+
+  /* stylelint-disable selector-class-pattern */
+  .react-pdf__Document .react-pdf__Page {
+    box-shadow: 0 0 0.5rem rgb(0, 0, 0, 25%);
+    margin: 0 auto;
+    width: min-content;
 
-  button {
-    padding: 0.25rem 0.5rem;
+    &:not(:last-child) {
+      margin-bottom: 1rem;
+    }
   }
 `;
 
-const DebugPanel = styled.div`
-  background: ${({ theme }) => theme.colors.containerLow};
-  color: ${({ theme }) => theme.colors.onBackgroundMuted};
-  border: 1px solid ${({ theme }) => theme.colors.outline};
-  padding: 0.5rem;
-  border-radius: 0.25rem;
-  gap: 0.5rem;
+const PdfControls = styled.div`
+  position: sticky;
+  left: 0;
+  top: 0;
+  z-index: 1;
+  width: 100%;
+  padding: 0.25rem 1rem;
+  background-color: ${(p) => p.theme.colors.inverseContainer};
+  color: ${(p) => p.theme.colors.onInverse};
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
 `;
 
-function DebugPanelToggleButton({
-  debugFlags,
-  setDebugFlags,
-  showDebugPanel,
-  setShowDebugPanel,
-}: {
-  debugFlags: DebugFlags;
-  setDebugFlags: (debugFlags: DebugFlags) => void;
-  showDebugPanel: boolean;
-  setShowDebugPanel: (showDebugPanel: boolean) => void;
-}) {
+const ZOOM_STEP = 0.25;
+const MIN_ZOOM = 0.25;
+const MAX_ZOOM = 2;
+
+function PdfViewer({ pdfData }: { pdfData?: Buffer }) {
+  const [numPages, setNumPages] = useState<number>();
+  const [zoom, setZoom] = useState(1);
+  const [currentPage, setCurrentPage] = useState(1);
+  const file = useMemo(() => ({ data: pdfData }), [pdfData]);
+
+  function onScroll(e: React.UIEvent<HTMLDivElement>) {
+    if (!numPages) return;
+    const { scrollHeight, scrollTop } = e.currentTarget;
+    // Add a fraction of the page height to the scroll position to make the
+    // transition happen a little earlier (when most of the next page is
+    // visible)
+    const pageHeight = scrollHeight / numPages;
+    const scrollProgress = (scrollTop + pageHeight / 6) / scrollHeight;
+    setCurrentPage(Math.floor(scrollProgress * numPages) + 1);
+  }
+  const loading = (
+    <Row
+      style={{
+        justifyContent: 'center',
+        alignItems: 'center',
+        fontSize: '3rem',
+        height: '100%',
+      }}
+    >
+      <Icons.Loading />
+    </Row>
+  );
+
   return (
-    <DebugPanelWrapper>
-      {showDebugPanel ? (
-        <DebugPanel>
-          <Row
-            style={{
-              justifyContent: 'space-between',
-              alignItems: 'center',
-            }}
+    <PdfContainer>
```

Comment:
I could easily see @adghayes wanting to use this in #4552. Coordinate on extracting for reuse?

Reply:
I don't think zoom is necessary in VxAdmin, and the location of the controls as a top bar might make less sense. But consistency is good. Thinking about it.

Reply:
@adghayes and I discussed and we will look into reusing the a shared PDF viewer component as a follow up